3d_armor_stand/init.lua

```diff
@@ -149,236 +149,137 @@ minetest.register_node("3d_armor_stand:top", {
 	tiles = {"blank.png"},
 })
 
-minetest.register_node("3d_armor_stand:armor_stand", {
-	description = S("Armor Stand"),
-	drawtype = "mesh",
-	mesh = "3d_armor_stand.obj",
-	tiles = {"3d_armor_stand.png"},
-	use_texture_alpha = "clip",
-	paramtype = "light",
-	paramtype2 = "facedir",
-	walkable = false,
-	selection_box = {
-		type = "fixed",
-		fixed = {
-			{-0.25, -0.4375, -0.25, 0.25, 1.4, 0.25},
-			{-0.5, -0.5, -0.5, 0.5, -0.4375, 0.5},
-		},
-	},
-	groups = {choppy=2, oddly_breakable_by_hand=2},
-	is_ground_content = false,
-	sounds = default.node_sound_wood_defaults(),
-	on_construct = function(pos)
-		local meta = minetest.get_meta(pos)
-		meta:set_string("formspec", armor_stand_formspec)
-		meta:set_string("infotext", S("Armor Stand"))
-		local inv = meta:get_inventory()
-		for _, element in pairs(elements) do
-			inv:set_size("armor_"..element, 1)
-		end
-	end,
-	can_dig = function(pos, player)
-		local meta = minetest.get_meta(pos)
-		local inv = meta:get_inventory()
-		for _, element in pairs(elements) do
-			if not inv:is_empty("armor_"..element) then
-				return false
-			end
+local function register_armor_stand(def)
+	local function owns_armor_stand(pos, meta, player)
+		if (def.name == "locked_armor_stand" and not has_locked_armor_stand_privilege(meta, player)) or
+		   (def.name == "shared_armor_stand" and (not minetest.is_player(player) or
+		   minetest.is_protected(pos, player:get_player_name()))) then
+			return false
```

Comment:
This is impractical to read. How about splitting individual cases into separate conditions?
```suggestion
		if def.name == "locked_armor_stand" and not has_locked_armor_stand_privilege(meta, player) then
			return false
		end
		local has_access = core.is_player(player) and not core.is_protected(pos, player:get_player_name())
		if def.name == "shared_armor_stand" and not has_access then
			return false
		end
```